test/lib/ufe/testParentCmd.py

```diff
@@ -386,6 +386,280 @@ def testParentAbsoluteSingleMatrixOp(self):
+                self.assertIn('Capsule1', childrenNames(xformChildren))
+            
+                # Confirm that the capsule has not moved in world space.  Must
+                # re-create the prim and its scene item after path change.
+                capsulePath = ufe.Path(
+                    [proxyShapePathSegment, 
+                     usdUtils.createUfePathSegment('/Xform1/Capsule1')])
+                capsulePrim = mayaUsd.ufe.ufePathToPrim(
+                    ufe.PathString.string(capsulePath))
+                capsuleXformable = UsdGeom.Xformable(capsulePrim)
+                capsuleItem = ufe.Hierarchy.createItem(capsulePath)
+                capsuleT3d = ufe.Transform3d.transform3d(capsuleItem)
+                capsuleWorld = capsuleT3d.inclusiveMatrix()
+                assertVectorAlmostEqual(
+                    self, capsuleWorldPre, matrixToList(capsuleWorld))
+            
+                # No change in the capsule's transform ops.
+                self.assertEqual(
+                    capsuleXformable.GetXformOpOrderAttr().Get(), 
+                    Vt.TokenArray(matrixOps))
+
+                # Matrix ops that were not targeted did not change.
+                for checkMatrixOp in matrixOps:
+                    if checkMatrixOp == matrixOp:
+                        continue
+                    op = UsdGeom.XformOp(
+                        capsulePrim.GetAttribute(checkMatrixOp))
+                    self.assertEqual(
+                        op.GetOpTransform(mayaUsd.ufe.getTime(
+                            ufe.PathString.string(capsulePath))),
+                        matrixValues[checkMatrixOp])
+
+            checkParentDone()
+
+            # Undo: the xform no longer has a child, the capsule is still where
+            # it has always been.
+            cmds.undo()
+
+            xformChildren = xformHier.children()
+            self.assertEqual(len(xformChildren), 0)
+
+            capsulePath = ufe.Path(
+                [proxyShapePathSegment, usdUtils.createUfePathSegment('/Capsule1')])
+            capsulePrim = mayaUsd.ufe.ufePathToPrim(
+                ufe.PathString.string(capsulePath))
+            capsuleXformable = UsdGeom.Xformable(capsulePrim)
+            capsuleItem = ufe.Hierarchy.createItem(capsulePath)
+            capsuleT3d = ufe.Transform3d.transform3d(capsuleItem)
+            capsuleWorld = capsuleT3d.inclusiveMatrix()
+            assertVectorAlmostEqual(
+                self, capsuleWorldPre, matrixToList(capsuleWorld))
+            self.assertEqual(
+                capsuleXformable.GetXformOpOrderAttr().Get(), 
+                Vt.TokenArray(matrixOps))
+
+            # Redo: capsule still hasn't moved.
+            cmds.redo()
+
+            checkParentDone()
+
+            # Go back to initial conditions for next iteration of loop.
+            cmds.undo()
+
+        # Restore initial conditions.
+        del os.environ['MAYA_USD_MATRIX_XFORM_OP_NAME']
```

Comment:
How about we spare other developers debugging and put an assert in setUp to detect when another test left this env variable set?